src/input_device_monitor/sender/concrete/socket_client_sender.rs

```diff
@@ -1,11 +1,92 @@
+use colored::Colorize;
+use dotenv_codegen::dotenv;
 use nannou::event::WindowEvent;
+use serde_json;
+use std::io::prelude::*;
+use std::os::unix::net::UnixStream;
+
+use crate::input_device_monitor::my_event::MyWindowEvent;
 use crate::input_device_monitor::sender::IEventSender;
 
-pub struct SocketClientSender;
+pub const SOCKET_SERVER_PATH: &str = dotenv!("SOCKET_SERVER_PATH");
+
+pub struct SocketClientSender {
+    stream: UnixStream,
+}
+
+impl SocketClientSender {
+    pub fn new(socket_path: &str) -> std::io::Result<Self> {
+        match UnixStream::connect(socket_path) {
+            Ok(stream) => Ok(Self { stream }),
+            Err(err) => {
+                println!(
+                    "{} {}\n",
+                    "Failed to connect to the socket server:".bold().red(),
+                    err
+                );
+                Err(err)
+            }
+        }
+    }
+}
 
 impl IEventSender for SocketClientSender {
-    fn send_event(&self, event: &WindowEvent) {
-        // Here is the implementation for SocketClientSender
-        println!("Sending event: {:?}", event);
+    fn send_event(&mut self, event: &WindowEvent) {
+        let my_event: MyWindowEvent = event.clone().into();
+        let event_json = serde_json::to_string(&my_event).unwrap();
+        writeln!(self.stream, "{}", event_json).unwrap();
+        println!("{} : {}", "✓ Sent".bold().green(), event_json);
```

Comment:
The colored printing looks great. The [execution script](https://tree.taiga.io/project/joseluis-teran-coffeetime/us/176) is redirecting stdout to a log file. Could this affect how it prints inside the log file?

Reply:
you're right, I think that this feedback isn't necessary for the log file, but yes for the log errors like when the socket server isn't running. So I created this validation in another module for these cases

Reply:
nice